### PR TITLE
166 implement UI for searching tags

### PR DESCRIPTION
**Pull request by tcoro**
Implement the search bar according to the [figma](https://www.figma.com/file/RevVhXP9iyGxmch31H0Tce/Project-Leek?node-id=115%3A341) mockup:

- [x] create search bar
- [x] implement into tag overview
- [x] implement functionality of filtering

### CODE REVIEW

File: packages/app/src/components/add-tag/AddTagStepTrack.vue [hunk 1]
```diff
@@ -110,6 +110,8 @@ export default defineComponent({
       const tagCopy = tag.value;
       tagCopy.spotifyTrackUri = track.uri;
       tagCopy.imageUrl = track.imageUri;
+      tagCopy.trackTitle = track.title;
+      tagCopy.trackArtists = track.artists.join(', ');
```
Comment (schnickalodeon) [inline packages/app/src/components/add-tag/AddTagStepTrack.vue:114]: Later we can (should change this)
In the PR #224 Implement UI for editing a tag I added a service to get the 

- title
- Artists

by the spotify uri

File: packages/app/src/components/uiBlocks/TagSearchResult.vue [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+  },
+
+  emits: {
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    'tag-selected': (_payload: NFCTag): boolean => true,
+  },
+
+  setup(props) {
+    const allTags = ref<NFCTag[]>([]);
+    const searchResult = ref<NFCTag[]>([]);
```
Comment (schnickalodeon) [inline packages/app/src/components/uiBlocks/TagSearchResult.vue:59]: Later we can (should change this)
In the PR #224 Implement UI for editing a tag I added a service to get the 

Then:

 const selectedTrack = ref<SpotifyTrack>();

File: packages/commons/src/nfc-tag.class.ts [hunk 3]
```diff
@@ -9,6 +9,10 @@ export default class NFCTag extends AbstractEntity {
 
   name!: string;
 
+  trackTitle!: string;
+
+  trackArtists!: string;
+
```
Comment (schnickalodeon) [inline packages/commons/src/nfc-tag.class.ts:15]: Later remove this

**Review by schnickalodeon — APPROVED** [review]

**Review by julianhahn — CHANGES_REQUESTED** [review]
something is wrong with the overflow on the right side. Please fix this

image

**Comment by tcoro** [comment]
> something is wrong with the overflow on the right side. Please fix this
> 
> image

I tried to replicate your problem and did not find the problem?
And just with looking at the code I couldnt find the possible line which produces this problem 
image

**Comment by tcoro** [comment]
@julianhahn it now looks like this:
image

**Review by julianhahn — APPROVED** [review]

**Pull request merged**